src/utils/CalendarUtils/CalendarUtils.js

```diff
@@ -37,30 +38,67 @@ export const elementExists = selector => {
   return javaScriptUtils.isDefined(element);
 };
 
-export const filterContributionDays = (dailyData, timeframe) => {
-  const isDay = dailyData.attributes.class === "day";
-
-  // Weekdays and months displayed around the calendar should be disregarded.
-  if (!isDay) {
-    return false;
-  }
-
+export const filterByTimeframe = (dailyDataWithContributions, timeframe) => {
   if (!timeframe.start && !timeframe.end) {
-    return true;
+    return dailyDataWithContributions;
   }
 
-  const contributionsDate = new Date(dailyData.attributes["data-date"]);
+  const filtered = Object.entries(dailyDataWithContributions).filter(
+    ([date]) => {
+      const convertedDate = new Date(date);
 
-  if (timeframe.start && !timeframe.end) {
-    return contributionsDate >= new Date(timeframe.start);
-  }
+      if (timeframe.start && !timeframe.end) {
+        return convertedDate >= new Date(timeframe.start);
+      }
 
-  if (!timeframe.start && timeframe.end) {
-    return contributionsDate <= new Date(timeframe.end);
-  }
+      if (!timeframe.start && timeframe.end) {
+        return convertedDate <= new Date(timeframe.end);
+      }
 
-  return (
-    contributionsDate >= new Date(timeframe.start) &&
-    contributionsDate <= new Date(timeframe.end)
+      return (
+        convertedDate >= new Date(timeframe.start) &&
+        convertedDate <= new Date(timeframe.end)
+      );
+    }
+  );
+
+  return filtered.reduce(
+    (result, [date, contributions]) => ({
+      ...result,
+      [date]: contributions
+    }),
+    {}
   );
 };
+
+export const aggregateCalendars = produce(
+  (draft, dailyDataWithContributions) => {
+    // Extract the weeks from the calendar.
+    const weeks = draft.children[0].children.slice(0, 53);
```

Comment:
This comment can very easily be replaced by a function. Always think if comments are really necessary.

Reply:
https://github.com/c-hive/team-contribution-calendar/pull/70/commits/b7853b76f8e8b671f1f5c647ce2f0155267976a2

Reply:
> Always think if comments are really necessary.

Thanks for pointing this out!